azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -194,6 +195,10 @@ public FlowRunnerManager(Props props, ExecutorLoader executorLoader,
             AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
             JobTypeManager.DEFAULT_JOBTYPEPLUGINDIR), globalProps,
             parentClassLoader);
+
+    executionMonitorManager = new FlowExecutionMonitorManager();
```

Comment:
nit:
This class can be Guice injected. I leave it to you if you wanna do this as part of this change. This is exactly what I would refer to as a class 'dependency'